### PR TITLE
Lf 2427 plain react - jumper menu makes the whole app re render on open

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-2427

This is just another approach to handle the performance issue using plain react and improving some function calls.

### CODE REVIEW

File: packages/dapp/src/components/Widget/Widget.tsx [hunk 1]
```diff
@@ -1,7 +1,166 @@
...
+
+  const widgetConfig: WidgetConfig = useMemo(() => {
+    let rpcs = {};
+    try {
+      rpcs = JSON.parse(apiUrl.VITE_CUSTOM_RPCS);
+    } catch (e) {
+      if (apiUrl.DEV) {
+        console.warn('Parsing custom rpcs failed', e);
+      }
+    }
```
Comment (chybisov) [inline packages/dapp/src/components/Widget/Widget.tsx:43]: There is some code duplication 🤔

File: packages/dapp/src/components/Widget/Widget.tsx [hunk 2]
```diff
@@ -1,7 +1,166 @@
...
+import { HiddenUI, LiFiWidget, WidgetConfig } from '@lifi/widget';
+import { useTheme } from '@mui/material/styles';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useUserTracking } from '../../hooks';
+import { useWallet } from '../../providers/WalletProvider';
+import { LanguageKey } from '../../types';
 
 export function Widget({ starterVariant }) {
-  const widgetConfig = useWidgetConfig((starterVariant = { starterVariant }));
```
[the remark this answers is not in the document]
Comment (dennyscode) [reply packages/dapp/src/components/Widget/Widget.tsx:5]: Let´s forget about this, the zustand one will take over that chunk as well

**Comment by dennyscode** [comment]
This was just a draft and should have been opened as a Draft Request, I suppose. Closing now as Zustand will be the state management tool for us to use. So, this code is deprecated. Go here: https://github.com/lifinance/jumper.exchange/pull/497

**Pull request closed without merging**